### DISCUSSION TITLE
Understand where LogProcessorFunction is involved ?

### DISCUSSION THREAD

**Discussion by fpasomeillan** [CloudSnorkel/CloudWatch2S3#22] [Q&A] [answered]
I've worked on setting up this template in my environment and I see everything is working as far as logs are exporting to S3, now , I may be missing something obvious here, but where is LogProcessorFunction ever called by anything else? I can't find any reference to it in kinesis or the CloudWatch event or the subscriber function, so why is this function needed and where is it hooked up ?

**Comment by kichik** [comment]
It is used to decompress the raw files arriving as JSON gzip.

https://github.com/CloudSnorkel/CloudWatch2S3/blob/5321db6bca2079446ef7fe47e3fd7441361429d3/CloudWatch2S3.template#L353-L373

**Reply by fpasomeillan** [reply]
I see, but this only happens if you enable 
![image](https://github.com/CloudSnorkel/CloudWatch2S3/assets/69057182/2701b498-16af-42d3-b915-9580dce40624)
right ?

**Reply by kichik** [reply, marked as answer]
Yes.

**Reply by fpasomeillan** [reply]
okay, I'm glad I'm not crazy, I didn't have that enabled originally, nor did I see a spot in the CF template to set it up, so I was trying to understand how everything talks to each other and never saw where it was ever using that function, thank you so much!
